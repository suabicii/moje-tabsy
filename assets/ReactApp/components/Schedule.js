import React, {useEffect, useState} from "react";
import {useSelector} from "react-redux";
import {sortedDrugsSelector} from "../features/drugs/drugsSlice";
import dayjs from "dayjs";
import DosingMoments from "./DosingMoments";

function Schedule() {
    const drugList = useSelector(sortedDrugsSelector);
    const currentDate = dayjs();
    const [drugListReduced, setDrugListReduced] = useState([]);
    const [dosingMomentsContent, setDosingMomentsContent] = useState([]);

    const checkIfCurrentTimeIsBeforeDosingMoment = (hour, minute) => {
        return currentDate.isBefore(currentDate.hour(parseInt(hour)).minute(parseInt(minute)), 'minute');
    };

    useEffect(() => {
        drugList.forEach(drug => {
            const dosingMoments = Object.entries(drug.dosingMoments);

            for (const [key, value] of dosingMoments) {
                const [hour, minute] = value.split(':');
                if (checkIfCurrentTimeIsBeforeDosingMoment(hour, minute)) {
                    setDrugListReduced([...drugListReduced, drug]);
                }
            }
        });
    }, []);

    return (
        <>
            {drugListReduced.length > 0 &&
                <ul className="list-unstyled">
                    {drugListReduced.map(({dosing, dosingMoments, id, name, unit}, index) => (
                        <div key={`${name}${id}`}>
                            <strong
                                data-testid={`schedule-drugName${id}`}>{name} </strong> â€“ {dosing} {unit}:
                            <DosingMoments content={dosingMoments} drugId={id}/>
                        </div>
                    ))}
                </ul>
            }
        </>
    );
}

export default Schedule;