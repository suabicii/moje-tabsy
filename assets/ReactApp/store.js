import {configureStore} from "@reduxjs/toolkit";
import drugsSlice from "./features/drugs/drugsSlice"
import userSlice from "./features/user/userSlice";
import darkModeSlice from "./features/darkMode/darkModeSlice";

export default configureStore({
    reducer: {
        drugs: drugsSlice,
        user: userSlice,
        darkMode: darkModeSlice
    }
});