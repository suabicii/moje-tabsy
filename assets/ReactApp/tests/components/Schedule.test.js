/**
 * @jest-environment jsdom
 * */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import Schedule from "../../components/Schedule";
import {Provider} from "react-redux";
import store from "../../store";
import Mockdate from "mockdate";

beforeAll(() => {
    Mockdate.set('2020-01-01');
});

afterAll(() => {
    Mockdate.reset();
});

it('should correctly render Schedule component', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
        <Provider store={store}>
            <Schedule/>
        </Provider>
    );

    expect(renderer.getRenderOutput()).toMatchSnapshot();
});