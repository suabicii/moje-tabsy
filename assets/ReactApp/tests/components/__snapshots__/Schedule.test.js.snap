// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Schedule component 1`] = `
<Context.Provider
  value={
    Object {
      "getServerState": undefined,
      "noopCheck": "once",
      "stabilityCheck": "once",
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Schedule />
</Context.Provider>
`;
